mods/rpvoicechat/src/Utils/PlayerAudioSource.cs

```diff
@@ -87,12 +93,49 @@ public void UpdatePlayer(float dt)
         if (player.Entity == null || player.Entity.SidedPos == null)
             return;
 
-        if (IsMuffled)
+        // If the player is on the other side of something to the listener, then the player's voice should be muffled
+        BlockSelection blocks = new BlockSelection();
+        EntitySelection entities = new EntitySelection();
+        capi.World.RayTraceForSelection(player.Entity.Pos.XYZ, capi.World.Player.Entity.Pos.XYZ, ref blocks, ref entities);
+        if (blocks != null)
+        {
+            int blockHitboxSize = 0;
+            foreach (Cuboidf val in blocks.Block.CollisionBoxes)
+            {
+                blockHitboxSize += (int) (val.Length * val.Height * val.Width);
+            }
+
+            capi.Logger.Debug("Total hitbox size: " + blockHitboxSize);
+
+            if(lowpassFilter == null)
+                lowpassFilter = new FilterLowpass(EffectsExtension, source);
+
+            lowpassFilter.Start();
+            //lowpassFilter.SetHFGain(Math.Max((float) 1 - (blockHitboxSize / 10),(float) 0));
```

Comment:
maybe instead of using this algo, you can use simply the number of blocks in between the players for now. so number of elements of blocks.Block.CollisionBoxes. Not necessarily needed. just an idea.